### PR TITLE
[Swap] Setting max swap amount missing substracting estimated fee

### LINKED ISSUE #1754: MAX button ends with Insufficient funds
## Prerequisites
Some assets for swapping or sending

## Expected Behavior
The maximum amount of the asset that can be transferred, minus the fees should be filled into the amount field.

## Current Behavior
The total amount of asset in the wallet is filled into the field, without accounting for gas, so if this amount is submitted, the tx will fail due to insufficient funds.

## Steps to Reproduce

https://user-images.githubusercontent.com/45055276/132924003-97ecf8c3-1974-4f76-980a-f5f085bfc1c3.mp4

1. Swap or send some asset
2. Click MAX button for amount
3. Submit transaction
4. Insufficient Funds quickly appears

## Possible Solution
Subtract required gas from total amount in wallet and fill that into the field instead.

## Your Environment - Release Version (skip if not using a release version)
V0.3.14 on Multiple OS

## Discussion at Discord
https://discord.com/channels/838986635756044328/839002220942721024/885209225519828992

### PR DESCRIPTION
**Pull request by gromxyz**
Closes #1754

### CODE REVIEW

File: src/renderer/components/swap/Swap.tsx [hunk 1]
```diff
@@ -146,7 +145,7 @@ export const Swap = ({
 }: SwapProps) => {
   const intl = useIntl()
 
-  const unlockedWallet = useMemo(() => isLocked(keystore) || !hasImportedKeystore(keystore), [keystore])
+  const unlockedWallet = useMemo(() => !isLocked(keystore) && hasImportedKeystore(keystore), [keystore])
```
Comment (veado) [inline src/renderer/components/swap/Swap.tsx]: Why to reverse the logic and not just rename this function to `lockedWallet`? _I know it's still WIP..._

Comment (gromxyz) [reply src/renderer/components/swap/Swap.tsx]: Yeah, that would actually be good too, will change it back.

File: src/renderer/components/swap/Swap.utils.test.ts [hunk 2]
```diff
@@ -651,4 +652,20 @@ describe('components/swap/utils', () => {
       expect(eqBaseAmount.equals(result, assetToBase(assetAmount(0.01525, inAssetDecimal)))).toBeTruthy()
     })
   })
+
+  describe('maxAmountToSwapMax1e8', () => {
+    it('returns max. balance to swap - estimated fees', () => {
+      const params = {
+        inputBalanceAmount: baseAmount(1000),
```
Comment (veado) [inline src/renderer/components/swap/Swap.utils.test.ts:659]: Can you add another tests, where `decimal` is `18` (for example for ETH)

File: src/renderer/components/swap/Swap.tsx [hunk 3]
```diff
@@ -501,11 +499,8 @@ export const Swap = ({
   // depends on users balances of source asset
   // Decimal always <= 1e8 based
   const maxAmountToSwapMax1e8: BaseAmount = useMemo(() => {
-    // make sure not logged in user can play around with swap
-    if (unlockedWallet) return assetToBase(assetAmount(Number.MAX_SAFE_INTEGER, sourceAssetAmountMax1e8.decimal))
-
-    return sourceAssetAmountMax1e8
-  }, [unlockedWallet, sourceAssetAmountMax1e8])
+    return Utils.maxAmountToSwapMax1e8(sourceAssetAmountMax1e8, swapFees)
```
Comment (veado) [inline src/renderer/components/swap/Swap.tsx]: Don't remove the case to handle locked / notimported wallet users - they still can play around.

File: src/renderer/components/swap/Swap.utils.ts [hunk 4]
```diff
@@ -257,3 +257,13 @@ export const minBalanceToSwap = (swapFees: Pick<SwapFees, 'inFee'>): BaseAmount
   // Over-estimate balance by 50%
   return feeToCover.times(1.5)
 }
+
+export const maxAmountToSwapMax1e8 = (assetAmount: BaseAmount, swapFees: Pick<SwapFees, 'inFee'>): BaseAmount => {
```
Comment (veado) [inline src/renderer/components/swap/Swap.utils.ts]: `inFee: BaseAmount` - no need to push down a "complex" object

**Review by veado — CHANGES_REQUESTED** [review]

File: src/renderer/components/swap/Swap.utils.test.ts [hunk 5]
```diff
@@ -651,4 +652,20 @@ describe('components/swap/utils', () => {
...
+        fees: {
+          inFee: {
+            amount: baseAmount(100),
```
Comment (veado) [inline src/renderer/components/swap/Swap.utils.test.ts]: Can you add another tests, where `decimal` is `18` (for example for ETH)

File: src/renderer/components/swap/Swap.utils.ts (hunk 4 above)
Comment (veado) [reply src/renderer/components/swap/Swap.utils.ts]: Pls change `assetAmount` -> `assetAmountMax1e8` to show it's max. `1e8` based

File: src/renderer/components/swap/Swap.utils.ts [hunk 6]
```diff
@@ -257,3 +257,9 @@ export const minBalanceToSwap = (swapFees: Pick<SwapFees, 'inFee'>): BaseAmount
   // Over-estimate balance by 50%
   return feeToCover.times(1.5)
 }
+
+export const maxAmountToSwapMax1e8 = (assetAmount: BaseAmount, inFeeAmount: BaseAmount): BaseAmount => {
```
Comment (veado) [inline src/renderer/components/swap/Swap.utils.ts]: `assetAmountMax1e8: BaseAmount, feeAmount: BaseAmount`

File: src/renderer/components/swap/Swap.utils.ts [hunk 7]
```diff
@@ -257,3 +257,9 @@ export const minBalanceToSwap = (swapFees: Pick<SwapFees, 'inFee'>): BaseAmount
...
+  const estimatedFee = to1e8BaseAmount(inFeeAmount.times(1.5))
```
Comment (veado) [inline src/renderer/components/swap/Swap.utils.ts]: `to1e8BaseAmount` -> `max1e8BaseAmount` (because passed `assetAmount` is `max1e8` based)

**Review by veado — CHANGES_REQUESTED** [review]

File: src/renderer/components/swap/Swap.utils.ts (hunk 6 above)
Comment (veado) [reply src/renderer/components/swap/Swap.utils.ts]: Pls add a comment to this function what we expect as params here. Just to remember it in few months from now.

**Review by veado — APPROVED** [review]

**Pull request merged**